internal/website/content/howto/blob/data.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Store and Access Unstructured Data"
+date: 2019-03-21T08:42:51-07:00
+draft: true
+weight: 2
+---
+
+Once you have [opened a bucket][] for the storage provider you want, you can
+store and access data from them using the standard Go I/O patterns.
+
+[opened a bucket]: {{< ref "./open-bucket.md" >}}
+
+## Writing data to a bucket
+
+To write data to a bucket, you create a writer, write data to it, and then close
+the writer. The writer implements [`io.Writer`][], so you can use any functions
+that take in an `io.Writer` like `io.Copy` or `fmt.Fprintln`.
+
+```go
+// Open the key "foo.txt" for writing with the default options.
+w, err := bucket.NewWriter(ctx, "foo.txt", nil)
+if err != nil {
+    return w
+}
+_, writeErr := fmt.Fprintln(w, "Hello, World!")
+closeErr := w.Close()
+if writeErr != nil {
+    return writeErr
+}
+if closeErr != nil {
+    return closeErr
+}
+```
+
+If you want to cancel an in-progress write (perhaps because the source you are
+reading from had an error), you can cancel the `Context` you pass to the writer.
+
+```go
+// Create a cancelable context from the existing context.
+writeCtx, cancelWrite := context.WithCancel(ctx)
+defer cancelWrite()
+
+// Open the key "foo.txt" for writing with the default options.
+w, err := bucket.NewWriter(ctx, "foo.txt", nil)
+if err != nil {
+    return err
+}
+if _, err := fmt.Fprintln(w, "Hello, World!"); err != nil {
```

Comment:
Actually, in the case you describe with `io.Copy`, calling `Close` without cancelling the `Context` would perform the write. So that might be a decent example, but I'm not sure it conveys more information than the current example and introduces new elements.

I'm hesitant to avoid adding a goroutine since that adds a fair amount of complexity. Would adding a second print perhaps be evocative enough? That would demonstrate interruption of a write more directly. WDYT?

Reply:
Ah, I'm missing something: calling `Writer.Close` is committing the write, and you want to abort it. Sorry. But to be fair, that isn't really clear, especially since you say I still have to call Close. So exactly when will calling Close _not_ commit the write? I'm confused.

Reply:
Here are the two properties:

1) `Close` must always be called.
2) Canceling the `Context` given to `NewWriter` before calling `Close` will abort the whole write operation.

If you are confused, then that is a failure of the documentation. Any suggestion on what would have made it clearer for you?

Reply:
I didn't actually read the blob documentation, so let's leave aside whether or not that is adequate. Either way, here in your how-to, you should generously explain all this. Something like, "Normally, calling Close on a blob.Writer has two effects: it releases resources [on the client, the server or both? maybe it doesn't matter], and it tells the provider that we are finished writing, so the blob should be [brought into existence? materialized?] in the bucket. But say we find a problem while writing. We must still release resources, but we don't want the partially-written blob to exist."
yadda, yadda, yadda, cancel.

Reply:
That is helpful. I've reworded the section to emphasize closing even more and reworked the paragraph to include the information you have there. PTAL.

(And to be clear: I was not making a statement on the reference documentation. I was making the point that this document under review was failing to convey the information it needed to. I'm in total agreement that the explanation in this section should convey all the considerations without requiring a trip to reference.)

Reply:
Much better, thanks.